### ISSUE TITLE
Combining Evidence (product vs. weighted average or average)

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/vbet#10]
Philip, thanks for explaining on #4 how you [combine evidence](https://riverscapes.github.io/vbet/algorithm.html#step-4----combined-evidence) in the  [code](https://github.com/Riverscapes/sqlBRAT/blob/master/vbet.py#L150). Given that we're likely to add other lines of evidence (#9), its worth thinking about both the numbers that are produced (from 0 to 1) and method. You are using a product, which tends to lower the number much more so than an average. We have the same issue in the ecohydraulic models.  *Question*, as we add more lines of evidence, do you think we should:
- stick with product?
- add option of average?
- add option of weighted average?

**Comment by philipbaileynar** [comment]
I totally agree that this should change. 

I'll be honest that, as the lines of evidence stack up, a simple neural network is my preferred approach. 

In the short term I like a weighted average. Super easy to implement, interpret and teach.

**Comment by joewheaton** [comment]
> I'll be honest that, as the lines of evidence stack up, a simple neural network is my preferred approach.

I need to see (someday) an example of how we do this and that could be a cool option.

>In the short term I like a weighted average. Super easy to implement, interpret and teach.

Agreed.

**Comment by MattReimer** [comment]
I think all of this is interesting but let's make sure we don't end up implementing a path parser or something silly here. 

Maybe like doing a nearest neighbour approximation you get 3-4 choices of algorithm (bilieanr, nearest neighbour etc) and those have really well-defined parameter sets. 

See my comment on #9 about a config file. If we do this we will really be pushing what a command-line argument parser can handle. Probably we'll need something more.

**Comment by joewheaton** [comment]
on 

> Maybe like doing a nearest neighbour approximation you get 3-4 choices of algorithm (bilieanr, nearest neighbour etc) and those have really well-defined parameter sets.

This is not a spatial neighborhood operation... this is just on a cell-by-cell basis, how do we combine the normalized values (between 0 and 1). So, I think it can be pretty simple. I do think the config files makes some sense.

**Comment by MattReimer** [comment]
yeah, I only meant there could be 3-4 choices in a dropdown of how things get combined instead of some complicated math parser like we do for brat hydrology equations. You get an average or you get a weighted average or you get..... X method. Whatever.
